public/app/plugins/datasource/loki/datasource.test.ts

```diff
@@ -30,8 +49,65 @@ describe('LokiDatasource', () => {
     replace: (a: string) => a,
   } as unknown) as TemplateSrv;
 
+  describe('when running range query with fallback', () => {
+    let ds: LokiDatasource;
+    beforeEach(() => {
+      const customData = { ...(instanceSettings.jsonData || {}), maxLines: 20 };
+      const customSettings = { ...instanceSettings, jsonData: customData };
+      ds = new LokiDatasource(customSettings, backendSrv, templateSrvMock);
+      backendSrvMock.datasourceRequest = jest.fn(() => Promise.resolve(legacyTestResp));
+    });
+
+    test('should try latest endpoint but fall back to legacy endpoint if it cannot be reached', async () => {
+      const options = getQueryOptions<LokiQuery>({
+        targets: [{ expr: 'rate({job="grafana"}[5m])', refId: 'B' }],
+        exploreMode: ExploreMode.Metrics,
```

Comment:
Should probably be `ExploreMode.Logs` since you can't see the mode selector in legacy versions.